packages/flutter/lib/src/cupertino/search_field.dart

```diff
@@ -0,0 +1,326 @@
+  /// look. The inset values were determined using the comparison tool in
+  /// https://github.com/flutter/platform_tests/.
+  final EdgeInsets prefixInsets;
+
+  /// Sets the padding insets for the prefix.
+  ///
+  /// Defaults to padding that replicates the `UISearchTextField` prefix
+  /// look. The inset values were determined using the comparison tool in
+  /// https://github.com/flutter/platform_tests/.
+  final EdgeInsets suffixInsets;
+
+  /// Sets the suffix widget's icon.
+  ///
+  /// Defaults to the X-Mark [CupertinoIcons.xmark_circle_fill]. The suffix is
+  /// customizable so that users can override it with other options, like a
+  /// bookmark icon.
+  final Icon suffixIcon;
+
+  /// Dictates when the X-Mark (suffix) should be visible.
+  ///
+  /// Defaults to only on when editing.
+  final OverlayVisibilityMode suffixMode;
+
+  /// Sets the X-Mark (suffix) action.
+  ///
+  /// Defaults to clearing the text. The suffix action is customizable
+  /// so that users can override it with other functionality, that isn't
+  /// necessarily clearing text.
+  final VoidCallback? onSuffixTap;
+
+  @override
+  Widget build(BuildContext context) {
+    // The icon size will be scaled by a factor of the accessibility text scale,
+    // to follow the behavior of `UISearchTextField`.
+    final double scaledIconSize =
+        MediaQuery.textScaleFactorOf(context) * itemSize;
+
+    // If decoration was not provided, create a decoration with the provided
+    // background color and border radius.
+    final BoxDecoration decoration = this.decoration ??
+        BoxDecoration(
+          color: CupertinoDynamicColor.resolve(
+              backgroundColor ?? CupertinoColors.tertiarySystemFill, context),
+          borderRadius:
+              borderRadius ?? const BorderRadius.all(Radius.circular(9.0)),
+        );
+
+    final TextStyle placeholderStyle = this.placeholderStyle ??
+        TextStyle(
+          color: CupertinoDynamicColor.resolve(
+              CupertinoColors.secondaryLabel, context),
+        );
+
+    final IconThemeData iconThemeData = IconThemeData(
+        color: CupertinoDynamicColor.resolve(itemColor, context),
+        size: scaledIconSize);
+
+    final Widget prefix = Padding(
+      child: IconTheme(
+          child: const Icon(CupertinoIcons.search), data: iconThemeData),
+      padding: prefixInsets,
+    );
+
+    // TODO(DanielEdrisian): Replace [GestureDetector] with a [CupertinoButton].
+    // The reason why I went with [GestureDetector] was because
+    // [CupertinoButton] was messing with the content size of the entire
+    // search field. Must find a way to get it working with a button, so that
+    // touch-down has that fading effect.
+    final Widget suffix = Padding(
+      child: GestureDetector(
+        child: IconTheme(child: suffixIcon, data: iconThemeData),
+        onTap: onSuffixTap ??
+            () {
+              if (controller?.text != '') {
```

Comment:
ah that's a good catch @LongCatIsLooong. Though unfortunately this means it needs to be slightly more complicated. Since the "default" behavior is to pass in no controller, it will be null and the clear button will not do anything by default. 

I think we should then supply one to the CupertinoTextField so that we can actually control the content of the text and clear it. That unfortunately means that since there's state in that controller (i.e. we don't want to build a new one each frame and throw the old one away with the entered text), the whole thing needs to be a StatefulWidget then. 

See https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/cupertino/text_field.dart#L629. We'll have to generally replicate that behavior along with creating a local controller if not supplied in initState and handling restoration in restoreState.